test/e2e/framework/node/helper.go

```diff
@@ -185,18 +183,16 @@ func AddExtendedResource(ctx context.Context, clientSet clientset.Interface, nod
 	extendedResource := v1.ResourceName(extendedResourceName)
 
 	ginkgo.By("Adding a custom resource")
-	extendedResourceList := v1.ResourceList{
-		extendedResource: extendedResourceQuantity,
-	}
-	patchPayload, err := json.Marshal(v1.Node{
-		Status: v1.NodeStatus{
-			Capacity:    extendedResourceList,
-			Allocatable: extendedResourceList,
-		},
```

Comment:
so the problem seems to be how we are generating the patch payload , cc: @liggitt

Reply:
https://github.com/kubernetes/api/blob/22c1e39ad8694905cfaad52a473d81f230d61bcb/core/v1/types.go#L6108 has an empty KubeletEndpoint, but `omitempty` does not work for structs, only for struct pointers. Therefore we get the `"kubeletEndpoint": {"port": 0}` in the patch and that field then gets set to zero.

Perhaps replace with server-side-apply or generate the patch with plain old `fmt.Sprintf`?

Reply:
+1 to using patch still, but constructing the patch more correctly. JSON marshaling `map[string]any{"status":map[string]any{...}}` with just the exact status fields you want would be fine, and a little better than sprintf

Reply:
updated.